MTGToolbox.Abstract/IDeckRepository.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+
+namespace MTGToolbox.Abstract
+{
+    public interface IDeckRepository
+    {
+        IEnumerable<IDeck> GetDecks();
+        IDeck GetDeckById(int id);
```

Comment:
Since you seemed to have been big on using interfaces even on the models, I changed the repos' return types to IDeck and ICard to show that you can work entirely in the context of interfaces but it is not necessary. This could've also been `IDeckRepository<TDeck>` and the implementation would've been `DeckRepository : IDeckRepository<Deck>`